spec/models/client_spec.rb

```diff
@@ -23,6 +23,46 @@
     end
   end
 
+  describe "Client transfer" do
+    let!(:prefix)  { create(:prefix) }
+    let!(:prefixes)  { create(:client_prefix, client: client, prefix: prefix) }
+    let!(:prefixes_dos)  { create(:provider_prefix, provider: provider, prefix: prefix) }
+    let(:new_provider) { create(:provider, symbol: "QUECHUA") }
```

Comment:
I would be explicit about the member_type. You could also add tests for member_types that are not allowed, e.g. member_only.